packages/cli/src/cmds/beacon/initBeaconState.ts

```diff
@@ -187,6 +188,15 @@ export async function initBeaconState(
     // Convert to `Uint8Array` to avoid unexpected behavior such as `Buffer.prototype.slice` not copying memory
     stateBytes = new Uint8Array(stateBytes.buffer, stateBytes.byteOffset, stateBytes.byteLength);
     const anchorState = getStateTypeFromBytes(chainForkConfig, stateBytes).deserializeToViewDU(stateBytes);
+    // Validate genesis state root
+    const stateRoot = toRootHex(anchorState.hashTreeRoot());
+    const expectedRoot = getGenesisStateRoot(args.network || defaultNetwork);
+    if (expectedRoot !== null && stateRoot !== expectedRoot) {
+      throw new Error(`Genesis state root mismatch, {
+        expected: ${expectedRoot},
+        found: ${stateRoot},
+      }`);
```

Comment:
need to format this differently

```suggestion
      throw Error(`Genesis state root mismatch expected=${expectedRoot} received=${stateRoot}`)
```